### DISCUSSION TITLE
Using provisioner for identifier com.example.app for com.example.app.Notifee

### DISCUSSION THREAD

**Discussion by saintmalik** [codemagic-ci-cd/codemagic-docs#1969] [Q&A]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

React Native

### Steps to reproduce

We added a NotifeeNotificationService plugin which led to having bundle identifier "com.example.app.NotifeeNotificationService" Now signing hasn't been possible since, we've been getting errors

### Expected results

A successful build

### Actual results

 - Did not find provisioning profile matching bundle identifier "com.example.app.NotifeeNotificationService" for target "NotifeeNotificationService" [Debug] from project "devapp"  
 - Did not find provisioning profile matching bundle identifier "com.example.app.NotifeeNotificationService" for target "NotifeeNotificationService" [Release] from project "devapp" 
 - 
and a final error in the build ipa stage

❌  /Users/builder/clone/ios/app.xcodeproj: error: "NotifeeNotificationService" requires a provisioning profile. Select a provisioning profile in the Signing & Capabilities editor. (in target 'NotifeeNotificationService' from project 'devapp')

Failed to archive /Users/builder/clone/ios/app.xcworkspace

**Comment by saintmalik** [comment]
any help

**Comment by kulakarla** [comment]
Hey @saintmalik, do you mean that your projects bundle identifier changed because of it?

**Reply by saintmalik** [reply]
no I mean i have two project identifier now, and I expect that the provisioning profile of com.example.app should work for com.example.app.NotifeeNotificationService, but it's not and com.example.app.NotifeeNotificationService has been spitting errors that it can't find provisioning profile matching bundle identifier

**Reply by kulakarla** [reply]
Hi @saintmalik 

How did you install Notifee?
I followed the instructions [for React Native on their website](https://notifee.app/react-native/docs/installation), and the app built and ran, didn't reproduce the error.

Did you by chance add Notifee from Xcode?

**Reply by saintmalik** [reply]
@kulakarla thanks for your response, yes we added it to Xcode, can you help me with the process you took to install Notifee that worked on your side?

regards

**Reply by kulakarla** [reply]
@saintmalik 

I need to also make sure what intention do you have with the Notifee plugin.
I assume you followed [this guide](https://notifee.app/react-native/docs/ios/remote-notification-support) when you were adding the plugin. This is a guide on how you can combine Apple's Remote Notifications with Notifee.

If this is not needed on your side, you can undo the steps you made while following the aforementioned guide (e.g remove the content from Podfile, remove the target etc), and then follow the React Native Installation guide I linked to you in an earlier reply.

However, please confirm if this was intentional - then this signing problem is indeed reproducible and I'll start looking for fixes.

**Reply by saintmalik** [reply]
Thank you, yes we went with this [guide](https://notifee.app/react-native/docs/ios/remote-notification-support), it was intentional, can you help reproducible it to get a fix

**Reply by kulakarla** [reply]
Hi there once again @saintmalik!

So I reproduced the whole error and there is a rather simple fix for it!
In Xcode targets, I just changed the bundle identifier of target `NotifeeNotificationService` to match the projects bundle identifier.

For a quick example, my projects bundle identifier was `com.error.fix`. When I added Notifee, it's bundle identifier was `com.error.fix.NotifeeNotificationService`, since by default targets get assigned this type of bundle identifier, that's where the error stems from. Well, luckily the bundle identifiers are easily editable in Xcode. Just change target Notifee's bundle identifier to match your projects and your all good!

Let me know if you encounter any more problems. I do believe this should solve it though!

**Reply by rignaneseleo** [reply]
I've got the same problem with onesignal. changing the build name didn't work, but creating a provisional profile did the job.

**Comment by kulakarla** [comment]
I'm tenatively closing this thread due to inactivity.
